arch/xtensa/include/kernel_arch_func.h

```diff
@@ -111,7 +112,7 @@ static ALWAYS_INLINE void arch_cohere_stacks(struct k_thread *old_thread,
 	/* Dummy threads appear at system initialization, but don't
 	 * have stack_info data and will never be saved.  Ignore.
 	 */
-	if (old_thread->base.thread_state & _THREAD_DUMMY) {
+	if (!osz) {
```

Comment:
Not understanding this logic.  The idea of a DUMMY thread is that it has no stack at all, we're running with no current thread, and on the interrupt stack.  The interrupt stack is always local to a specific CPU and thus never needs to be flushed or invalidated.  Is this fix really required?  If so, I think the root problem must be somewhere else.

Reply:
This is needed because if `arch_cohere_stacks()` returns there, it doesn't set `EXCSAVE3`, then in `xtensa_switch` in arch/xtensa/core/xtensa-asm2-util.S it reads 0 from `EXCSAVE3` and flushes a huge address range, which also delays the secondary core start up.

Reply:
Right, but that's the protection the original code offered.  It ddn't flush dummy stacks.  The new code **will** try to flush a dummy stack, but will also refuse to flush a stack with a zero size.  Now... it's probably true that the stack size field of a dummy stack is zero (but AFAIK nothing in the architecture guarantees that, dummy thread objects exist as temporary stack data and I frankly don't know if they're fully initialized or not), so... maybe this is a noop.

But if it's a noop we should just skip it.  And if it's not we should be clear about exactly what behavior is desired.  To my eyes the original test is clearer.

Reply:
@andyross the problem is that it is very much *not* a NOP. And that is a bug. The fact, that secondary core takes 0 from `EXCSAVE3`, subtracts some valid address from it (or the other way round) and runs cache flushing on all those megabytes is a bug IMHO. Maybe my fix isn't perfect or even not correct. But I'd really like to fix this. The other option is in `xtensa_switch` to check `EXCSAVE3` for 0 and skip flushing in such a case? Would that be a correct solution?

Reply:
I'm still not understanding why you're getting a zero in that field for a non-dummy thread.  Something else is wrong if that's happening.  I just tried with a quick __ASSERT and didn't see this, how are you making it happen?

Reply:
@andyross sorry for not explaining properly. It **is** the dummy thread, running on a secondary core, using the IRQ stack.

Reply:
So what this change does it allows the dummy thread to proceed and set `EXCSAVE3` accordingly instead of bailing out on the `if` above